packages/solidity-contracts/contracts/fuelchain/FuelMessagePortal/v3/FuelMessagePortalV3.sol

```diff
@@ -93,6 +94,52 @@ contract FuelMessagePortalV3 is FuelMessagePortalV2 {
         _executeMessage(messageId, message);
     }
 
+    /// @notice Executes a message in the given header
+    /// @param messageId The id of message to execute
+    /// @param message The message to execute
+    function _executeMessage(bytes32 messageId, Message calldata message) internal virtual override nonReentrant {
+        if (_incomingMessageSuccessful[messageId]) revert AlreadyRelayed();
+
+        //set message sender for receiving contract to reference
+        _incomingMessageSender = message.sender;
+
+        // v2: update accounting if the message carries an amount
+        bool success;
+        bytes memory result;
+        if (message.amount > 0) {
+            uint256 withdrawnAmount = message.amount * PRECISION;
+
+            // Underflow check enabled since the amount is coded in `message`
+            totalDeposited -= withdrawnAmount;
+
+            (success, result) = address(uint160(uint256(message.recipient))).call{value: withdrawnAmount}(message.data);
+        } else {
+            (success, result) = address(uint160(uint256(message.recipient))).call(message.data);
+        }
+
+        if (!success) {
+            // Look for revert reason and bubble it up if present
+            if (result.length > 0) {
+                // The easiest way to bubble the revert reason is using memory via assembly
+                /// @solidity memory-safe-assembly
+                assembly {
+                    let returndata_size := mload(result)
+                    revert(add(32, result), returndata_size)
+                }
+            }
+            revert MessageRelayFailed();
+        }
+
+        //unset message sender reference
+        _incomingMessageSender = NULL_MESSAGE_SENDER;
```

Comment:
What's the benefit of resetting here?

Reply:
When a messag from L2 to L1 is relayed, this parameter is set to the sender temporarily.

Then a possible callback is made to a L1 contract (e.g., call the L1 bridge to release tokens)

The L1 called contract has the option to query the portal for who is the sender of the message (e.g., the L1 bridge queries the message portal to ensure that the incoming message sender is the L2 counterpart)

After L1 target contract has been called, we reset the variable. For the flow of the bridge itself this might not be strictly needed as we only accept one message at a time via callback, but there might other flows and implementations that do need to validate the correctness of this variable at a time other than the callback.